pkg/reconciler/vertex/controller.go

```diff
@@ -300,7 +301,6 @@ func (r *vertexReconciler) orchestratePods(ctx context.Context, vertex *dfv1.Ver
 		}
 	}
 
-	currentReplicas := int(vertex.Status.Replicas)
```

Comment:
This is the place to set `lastScaledAt `. Should not use current `vertex.Status.Replicas` as it has been updated above.